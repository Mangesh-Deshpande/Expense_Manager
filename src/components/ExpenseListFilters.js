import React from "react";
import { connect } from "react-redux";
import moment from "moment";
import "react-dates/initialize";
import { DateRangePicker } from "react-dates";
import {
  setTextFilter,
  sortByAmount,
  sortByDate,
  setStartDate,
  setEndDate
} from "../actions/filters";

// const id =moment().startOf('month').unix();
// console.log(id);

class ExpenseListFilters extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      calendarFocused: null
    };
  }
  onDatesChange = ({ startDate, endDate }) => {
    // this.setState({ startDate, endDate });
    // console.log(startDate);
    this.props.dispatch(setStartDate({ startDate }));
    this.props.dispatch(setEndDate({ endDate }));
  };

  onFocusChange = calendarFocused => {
    this.setState({ calendarFocused });
  };

  render() {
    return (
      <div>
        <input
          type="text"
          value={this.props.filters.text}
          onChange={e => {
            this.props.dispatch(setTextFilter(e.target.value));
            // console.log(e.target.value);
          }}
        />
        <select
          value={this.props.filters.sortBy}
          onChange={e => {
            if (e.target.value === "amount") {
              this.props.dispatch(sortByAmount());
            } else if (e.target.value === "date") {
              this.props.dispatch(sortByDate());
            }
            // console.log(e.target.value);
          }}
        >
          <option value="date">Date</option>
          <option value="amount">Amount</option>
        </select>

        <DateRangePicker
          startDate={this.props.filters.startDate}
          startDateId={
            !this.props.filters.startDate
              ? ""
              : this.props.filters.startDate.toString()
          }
          endDate={this.props.filters.endDate}
          endDateId={
            !this.props.filters.endDate
              ? ""
              : this.props.filters.endDate.toString()
          }
          onDatesChange={this.onDatesChange}
          focusedInput={this.state.calendarFocused}
          onFocusChange={this.onFocusChange}
          numberOfMonths={1}
          showClearDates={true}
          isOutsideRange={() => false}
          hideKeyboardShortcutsPanel={true}
        />
      </div>
    );
  }
}

const mapStateToProps = state => {
  return {
    filters: state.filters
  };
};

export default connect(mapStateToProps)(ExpenseListFilters);
