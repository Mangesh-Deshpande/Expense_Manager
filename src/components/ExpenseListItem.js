import React from 'react';
import { connect } from 'react-redux';
import {Link} from 'react-router-dom';
import moment from 'moment';


const ExpenseListItem = ({id,description,amount,createdAt}) =>{

   return (
    <div>
        <Link to={`/edit/${id}`} ><h3>{description} </h3></Link>
        <p>{amount/100} - {moment(createdAt).format('DD/MM/YY')} -
        {description}</p>
    </div>
   );
 };


export default ExpenseListItem;