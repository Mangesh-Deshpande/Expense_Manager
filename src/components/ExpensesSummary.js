import React from "react";
import { connect } from "react-redux";
import filteredExpenses from "../selectors/expenses";
import expensesTotal from "../selectors/expenses-total";
import numeral from "./NumeralforRupee";

export const ExpensesSummary = props => {
  const expenseWord = props.expense.count === 1 ? "expense" : "expenses";
  return (
    <div>
      <p>
        Viewing {props.expense.count} {expenseWord} totalling
        {numeral(props.expense.sum / 100).format("$0,0.00")}
      </p>
    </div>
  );
};

const mapStatetoProps = state => {
  const visibleExpenses = filteredExpenses(state.expenses, state.filters);
  return {
    expense: {
      sum: expensesTotal(visibleExpenses),
      count: visibleExpenses.length
    }
  };
};

export default connect(mapStatetoProps)(ExpensesSummary);
