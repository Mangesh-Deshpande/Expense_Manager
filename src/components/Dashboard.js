import React from "react";
import ExpenseList from "./ExpenseList";
import ExpenseListFilters from "./ExpenseListFilters";
import ExpensesSummary from "./ExpensesSummary";

const ExpensifyDashboardPage = () => (
  <div>
    <ExpenseListFilters />
    <ExpensesSummary />
    <ExpenseList />
  </div>
);

export default ExpensifyDashboardPage;
