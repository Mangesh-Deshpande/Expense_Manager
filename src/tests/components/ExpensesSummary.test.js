import React from "react";
import { ExpensesSummary } from "../../components/ExpensesSummary";
import { shallow } from "enzyme";

test("should render Expenses Summary correctly with one expense", () => {
  const expense = {
    count: 1,
    sum: 128900
  };
  const wrapper = shallow(<ExpensesSummary expense={expense} />);

  expect(wrapper).toMatchSnapshot();
});

test("should render Expenses Summary correctly with multiple expenses", () => {
  const expense = {
    count: 4,
    sum: 128343900
  };
  const wrapper = shallow(<ExpensesSummary expense={expense} />);

  expect(wrapper).toMatchSnapshot();
});
