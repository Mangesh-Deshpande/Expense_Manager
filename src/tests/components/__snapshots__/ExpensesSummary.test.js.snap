// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Expenses Summary correctly with multiple expenses 1`] = `
<div>
  <p>
    Viewing 
    4
     
    expenses
     totalling
    ₹1,283,439.00
  </p>
</div>
`;

exports[`should render Expenses Summary correctly with one expense 1`] = `
<div>
  <p>
    Viewing 
    1
     
    expense
     totalling
    ₹1,289.00
  </p>
</div>
`;
